pkg/resources/function_commons.go

```diff
@@ -0,0 +1,344 @@
+		importsDescription:            "The location (stage), path, and name of the file(s) to import, such as a JAR or other kind of file. The JAR file might contain handler dependency libraries. It can contain one or more .class files and zero or more resource files. JNI (Java Native Interface) is not supported. Snowflake prohibits loading libraries that contain native code (as opposed to Java bytecode). A non-JAR file might a file read by handler code. For an example, see [Reading a file specified statically in IMPORTS](https://docs.snowflake.com/en/developer-guide/udf/java/udf-java-cookbook.html#label-reading-file-from-java-udf-imports). Consult the [docs](https://docs.snowflake.com/en/sql-reference/sql/create-function#scala).",
+		packagesDescription:           "The name and version number of Snowflake system packages required as dependencies. The value should be of the form `package_name:version_number`, where `package_name` is `snowflake_domain:package`.",
+		handlerDescription:            "The name of the handler method or class. If the handler is for a scalar UDF, returning a non-tabular value, the HANDLER value should be a method name, as in the following form: `MyClass.myMethod`.",
+		targetPathDescription:         "The TARGET_PATH clause specifies the location to which Snowflake should write the compiled code (JAR file) after compiling the source code specified in the `function_definition`. If this clause is included, you should manually remove the JAR file when it is no longer needed (typically when the UDF is dropped). If this clause is omitted, Snowflake re-compiles the source code each time the code is needed. The JAR file is not stored permanently, and you do not need to clean up the JAR file. Snowflake returns an error if the TARGET_PATH matches an existing file; you cannot use TARGET_PATH to overwrite an existing file.",
+	}
+	sqlFunctionSchemaDefinition = functionSchemaDef{
+		additionalArguments:           []string{},
+		functionDefinitionDescription: functionDefinitionTemplate("SQL", "https://docs.snowflake.com/en/developer-guide/udf/sql/udf-sql-introduction"),
+	}
+)
+
+var (
+	javaFunctionSchema       map[string]*schema.Schema
+	javascriptFunctionSchema map[string]*schema.Schema
+	pythonFunctionSchema     map[string]*schema.Schema
+	scalaFunctionSchema      map[string]*schema.Schema
+	sqlFunctionSchema        map[string]*schema.Schema
+)
+
+// TODO [SNOW-1348103]: add null/not null
+// TODO [SNOW-1348103]: currently database and schema are ForceNew but based on the docs it is possible to rename with moving to different db/schema
+// TODO [SNOW-1348103]: copyGrants and orReplace logic omitted for now, will be added to the limitations docs
+// TODO [SNOW-1348103]: temporary is not supported because it creates a per-session object; add to limitations/design decisions
+var functionBaseSchema = map[string]schema.Schema{
```

Comment:
Nit: we can have this as a function to prevent accidental modifications.

Reply:
Changed in [the next PR](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3263)